### PR TITLE
Clean up and improve the `Poset` module in various ways

### PR DESCRIPTION
**Pull request by ayberkt**
I have done a couple of things.
- Cleaned up the definition poset isomorphism so that it coincides more directly with the categorical notion.
- Introduced a distinction between the type of monotonic poset equivalences and poset isomorphisms and proved them to be equivalent. These are denoted `≃ₚ` and `≅ₚ` respectively.
- Defined monotonic maps between posets and their composition.

I will finish #303 once this is merged.

### CODE REVIEW

File: Cubical/Structures/Poset.agda [hunk 1]
```diff
@@ -16,45 +18,83 @@ open import Cubical.Data.Sigma.Properties
...
-
 isSetOrder : (ℓ₁ : Level) (A : Type ℓ₀) → isSet (Order ℓ₁ A)
 isSetOrder ℓ₁ A = isSetΠ2 λ _ _ → isSetHProp
 
-Order-is-SNS : SNS {ℓ₀} (Order ℓ₁) order-iso
-Order-is-SNS {ℓ₁ = ℓ₁} {X = X}  _⊑₀_ _⊑₁_ = f , f-equiv
+-- We first start by defining what it means a for a function to be
+-- order-preserving. The name "monotonic" is reserved for partial orders.
+
+isOrderPreserving : (M : Σ (Type ℓ₀) (Order ℓ₁)) (N : Σ (Type ℓ₀′) (Order ℓ₁′))
```
Comment (mortberg) [inline Cubical/Structures/Poset.agda:32]: Wouldn't `M` and `N` look a bit nicer if you used `TypeWithStr` (https://github.com/agda/cubical/blob/master/Cubical/Foundations/Structure.agda#L14)?

Comment (ayberkt) [reply Cubical/Structures/Poset.agda:32]: Yes, I should get into a habit of using `TypeWithStr`.

Comment (ayberkt) [reply Cubical/Structures/Poset.agda]: Addressed in abd3a7f9867f495a0b5c3a259cf24984727439be.

**Comment by ayberkt** [comment]
@mortberg by the way, it might be a good idea to disallow working directly on `master`. The other PR I just made (#316) is failing the check because it was branched from a failing commit on `master`.

**Comment by ayberkt** [comment]
I suppose we would always like to avoid failing commits on `master`.

**Comment by mortberg** [comment]
I'm confused. Where did the failing commit to master come from?

**Comment by ayberkt** [comment]
> I'm confused. Where did the failing commit to master come from?

a65649653b04766583dbb179916d5846dc2bd464 is failing. I actually don't know where it comes from; I just assumed it was a result of working directly on master but maybe I was wrong?

**Comment by mortberg** [comment]
> > I'm confused. Where did the failing commit to master come from?
> 
> [a656496](https://github.com/agda/cubical/commit/a65649653b04766583dbb179916d5846dc2bd464) is failing. I actually don't know where it comes from; I just assumed it was a result of working directly on master but maybe I was wrong?

I just merged some things a bit too fast... The problem should be easy to fix: https://github.com/agda/cubical/pull/310

**Comment by ayberkt** [comment]
> I just merged some things a bit too fast... The problem should be easy to fix: #310

This should be fine to merge now.

**Pull request merged**